### PR TITLE
Updates for compat, and also fix keyword arg for jldopen

### PR DESCRIPTION
**Pull request by musm**
I believe this should fix 0.6 failures on JLD  due to keyword arg failure

### CODE REVIEW

File: test/v0.2.jl [hunk 1]
```diff
@@ -125,7 +125,7 @@ end
 
 for fn in ("v0.2.26.jld", "v0.2.28.jld")
     for mmap = (true, false)
-        fidr = jldopen(joinpath(splitdir(@__FILE__)[1], fn), "r", mmaparrays=mmap)
+        fidr = jldopen(joinpath(splitdir(@__FILE__)[1], fn), "r"; mmaparrays=mmap)
         @check fidr x
         @check fidr A
```
Comment (musm) [inline test/v0.2.jl]: See https://github.com/JuliaIO/JLD.jl/blob/master/src/JLD00.jl#L98

File: test/v0.2.jl [hunk 2]
```diff
@@ -60,7 +60,7 @@ rng = 1:5
 objwithpointer = r"julia"
 # Custom BitsType (#99)
 bitstype 64 MyBT
-bt = reinterpret(MyBT, 55)
```
Comment (musm) [inline test/v0.2.jl]: This needs to be `Int64` or it will fail on a 32 bit system

**Comment by musm** [comment]
Looks like tests pass now on nightly with this

**Comment by simonster** [comment]
FWIW, CI isn't set up to run on this repo, so if you want to check if this actually fixes the test failures, you will need to run it locally. I doubt it does; `f(..., kw=x)` and `f(...; kw=x)`are both valid ways of calling a function with keyword arguments, and AFAIK nothing has changed about that on master. Instead, I would guess it's a world age issue caused by loading the JLD00 module on demand, which is no longer allowed (at least the way JLD was doing it) since JuliaLang/julia#265.

**Comment by musm** [comment]
Yeah your probably 100 right. Fyi Travis will be fixed by this pr https://github.com/JuliaIO/JLDArchives.jl/pull/10

**Comment by musm** [comment]
Is this g2g @simonster ?

**Comment by musm** [comment]
nightly failures seem to be just as you predicted

**Comment by simonster** [comment]
Thanks. I think this is ready to merge. Can someone with commit access do that? We still need to figure out how to solve the JLD00 problem on master, but that's a change that needs to happen in JLD.

**Comment by timholy** [comment]
I added this repo to the HDF5/JLD team, so you should both be able to merge now.

**Pull request merged**